x-pack/apm-server/aggregation/servicemetrics/aggregator.go

```diff
@@ -268,13 +291,23 @@ func (mb *metricsBuffer) storeOrUpdate(key aggregationKey, metrics serviceMetric
 	entry := &mb.space[mb.entries]
 	entry.aggregationKey = key
 
-	entry.serviceMetrics = serviceMetrics{
-		transactionDuration: metrics.transactionDuration,
-		transactionCount:    metrics.transactionCount,
-		failureCount:        metrics.failureCount,
-		successCount:        metrics.successCount,
+	if entry.serviceMetrics.histogram == nil {
+		entry.serviceMetrics = serviceMetrics{
+			transactionDuration: metrics.transactionDuration,
+			transactionCount:    metrics.transactionCount,
+			failureCount:        metrics.failureCount,
+			successCount:        metrics.successCount,
```

Comment:
Why would you set `duration`, `count`, `failureCount` and `successCount` dependent of `histogram==nil`? Isn't that independent of the histogram initialization?

Reply:
Good catch! Updated! +1